common/lib/xmodule/xmodule/js/src/video/09_video_caption.js

```diff
@@ -110,12 +111,18 @@
                     '</div>'
                 ].join('');
 
-                var template = [
-                    '<div class="subtitles" role="region" id="transcript-' + this.state.id + '">',
-                        '<h3 id="transcript-label-' + this.state.id + '" class="transcript-title sr"></h3>',
-                        '<ol id="transcript-captions" class="subtitles-menu" lang="' + this.state.lang + '"></ol>',
-                    '</div>'
-                ].join('');
+                var template = HtmlUtils.interpolateHtml(
+                        HtmlUtils.HTML([
+                            '<div class="subtitles" role="region" id="transcript-{courseId}">',
+                                '<h3 id="transcript-label-{courseId}" class="transcript-title sr"></h3>',
+                                '<ol id="transcript-captions" class="subtitles-menu" lang="{courseLang}"></ol>',
+                            '</div>'
+                        ].join('')),
+                        {
+                            courseId: this.state.id,
+                            courseLang: this.state.lang
+                        }
+                    ).toString();
 
                 this.loaded = false;
                 this.subtitlesEl = $(template);
```

Comment:
I think it may be better to follow the pattern with `linkHtml` below, where `template` would be renamed to `subtitlesHtml`, and the `toString()` call would be dropped from the call above, and moved here as:
```
this.subtitlesEl = $(subtitlesHtml.toString());
```
This line could also be:
```
this.subtitlesEl = $(HtmlUtils.ensureHtml(subtitlesHtml).toString());
```
In this case there isn't much difference because these lines are so visually close, but `ensure()` makes sure that it will always be properly escaped, in this case, simply by verifying that it is already an Html Snippet.